.github/workflows/ci.yaml

```diff
@@ -0,0 +1,100 @@
+name: ci
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - actions
+jobs:
+  test:
+    name: test
+    env:
+      # Emit backtraces on panics.
+      RUST_BACKTRACE: 1
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        build:
+          # We test ripgrep on a pinned version of Rust, along with the moving
+          # targets of 'stable' and 'beta' for good measure.
+          - stable
+          - nightly
+          # Our release builds are generated by a nightly compiler to take
+          # advantage of the latest optimizations/compile time improvements. So
+          # we test all of them here. (We don't do mips releases, but test on
+          # mips for big-endian coverage.)
+          - macos
+          - win-msvc
+        include:
+          - build: stable
+            os: ubuntu-18.04
+            rust: stable
+          - build: macos
+            os: macos-latest
+            rust: stable
+          - build: win-msvc
+            os: windows-2019
+            rust: stable
+          - build: nightly
+            os: ubuntu-18.04
+            rust: nightly
```

Comment:
Right now on travis, our nightly branch is allowed to fail and we just use it as an FYI. Is there a way to do this here too?

Reply:
Not now unfortunately. I'm not familiar with setting up checks for pull requests, but I guess that we could do ci action with only stable as a gate for pull requests and ci_nightly which would build but would not guard anything.